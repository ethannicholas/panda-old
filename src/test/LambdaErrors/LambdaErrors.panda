method test1(f:(Int)=>(Real)) {
}

method test1(f:(Int)=>(String)) {
}

method test2(f:(Int)=>(Real)) {
}

method test2(f:(String)=>(Real)) {
}

def f1 := x => x * 2
def f2:(Int)=>(Int) := x => x.broken
test1(x => x.broken)
test2(x => x.startsWith("Hello"))