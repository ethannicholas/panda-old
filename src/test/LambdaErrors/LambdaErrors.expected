error: LambdaErrors.panda:13:11: lambda type cannot be determined from context
error: LambdaErrors.panda:14:30: unknown identifier
error: LambdaErrors.panda:15:1: no match found for method test1(<lambda>)
    possible error in lambda at LambdaErrors.panda:15:7:
        when interpreted as (panda.core.Int32)=>(...):
            unknown identifier at LambdaErrors.panda:15:13
error: LambdaErrors.panda:16:1: no match found for method test2(<lambda>)
    possible error in lambda at LambdaErrors.panda:16:7:
        when interpreted as (panda.core.Int32)=>(...):
            no match found for method startsWith(panda.core.String) at LambdaErrors.panda:16:12
        when interpreted as (panda.core.String)=>(...):
            lambda evaluated to 'panda.core.Bit', which cannot be used where 'panda.core.Real64' is expected
4 errors
